src/puyapy_mocks/_primatives.py

```diff
@@ -0,0 +1,386 @@
+        if isinstance(other, int):
+            return UInt64(other ^ self.__match_value__)
+        else:
+            return NotImplemented
+
+    def __ixor__(self, other: UInt64 | int) -> UInt64:  # noqa: PYI034
+        if isinstance(other, UInt64):
+            self.__match_value__ ^= other.__match_value__
+        elif isinstance(other, int):
+            self.__match_value__ ^= other
+        else:
+            return NotImplemented
+        return self
+
+    def __or__(self, other: UInt64 | int) -> UInt64:
+        if isinstance(other, UInt64):
+            return UInt64(self.__match_value__ | other.__match_value__)
+        elif isinstance(other, int):
+            return UInt64(self.__match_value__ | other)
+        else:
+            return NotImplemented
+
+    def __ror__(self, other: UInt64 | int) -> UInt64:
+        if isinstance(other, int):
+            return UInt64(other | self.__match_value__)
+        else:
+            return NotImplemented
+
+    def __ior__(self, other: UInt64 | int) -> UInt64:  # noqa: PYI034
+        if isinstance(other, UInt64):
+            self.__match_value__ |= other.__match_value__
+        elif isinstance(other, int):
+            self.__match_value__ |= other
+        else:
+            return NotImplemented
+        return self
+
+    def __invert__(self) -> UInt64:
+        return UInt64(~self.__match_value__)
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+    def __str__(self) -> str:
+        return f"{self.__match_value__}"
+
+    def __int__(self) -> int:
+        return self.__match_value__
+
+    def __index__(self) -> int:
+        return self.__match_value__
+
+    def __bytes__(self) -> bytes:
+        return int.to_bytes(self.__match_value__, 8)
+
+
+class Bytes:
+    __match_value__: bytes
+    __match_args__ = ("__match_value__",)
+
+    def __init__(self, value: bytes = b"", /) -> None:
+        self.__match_value__ = value
+
+    def __bytes__(self) -> bytes:
+        return self.__match_value__
+
+    @property
+    def length(self) -> UInt64:
+        return UInt64(len(self.__match_value__))
+
+    def __getitem__(self, index: UInt64 | int | slice) -> Bytes:
+        if isinstance(index, (UInt64, int)):
+            i = int(index)
+            return Bytes(self.__match_value__[i : i + 1])
+        elif isinstance(index, slice):
+            return Bytes(self.__match_value__[index.start : index.stop : index.step])
+        else:
+            return NotImplemented
+
+    def __add__(self, other: Bytes | bytes) -> Bytes:
+        if isinstance(other, bytes):
+            other = Bytes(other)
+        return Bytes(self.__match_value__ + bytes(other))
+
+    def __eq__(self, other: Bytes | bytes) -> bool:  # type: ignore[override]
+        return self.__match_value__ == bytes(other)
+
+    def __int__(self) -> int:
+        return int.from_bytes(self.__match_value__)
+
+    def __repr__(self) -> str:
+        return self.__repr__()
```

Comment:
@neilcampbell isn't this infinite recursion? I think this might need a str object instead

Reply:
It sure is. That should have been `.__str__()`